File: php/features.php
<?php
declare(strict_types=1);

// EsiDocumentation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EsiDocumentationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EsiDocumentationBaseFeature();
            case "test":
                return new EsiDocumentationTestFeature();
            default:
                return new EsiDocumentationBaseFeature();
        }
    }
}

File: py/features.py
# EsiDocumentation SDK feature factory

from feature.base_feature import EsiDocumentationBaseFeature
from feature.test_feature import EsiDocumentationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EsiDocumentationBaseFeature(),
        "test": lambda: EsiDocumentationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
